lib/reporting/authentication_report.rb

```diff
@@ -54,62 +54,72 @@ def progress?
+            sp_redirect_initiated_new_users,
+            format_as_percent(
+              numerator: sp_redirect_initiated_new_users,
+              denominator: email_confirmation,
+            ),
+          ],
+          [
+            'AAL2 Authentication Requests from Partner',
+            oidc_auth_request,
+            format_as_percent(numerator: oidc_auth_request, denominator: oidc_auth_request),
+          ],
+          [
+            'AAL2 Authenticated Requests',
+            sp_redirect_initiated_after_oidc,
+            format_as_percent(
+              numerator: sp_redirect_initiated_after_oidc,
+              denominator: oidc_auth_request,
+            ),
+          ],
+        ],
+      ]
+    end
+
+    def as_tables_with_options
+      as_tables.each_with_index.map do |table, index|
+        title = (index == 0) ? 'Overview' : 'Authentication Funnel Metrics'
+        table.unshift({ title: })
+      end
     end
 
-    # rubocop:disable Metrics/BlockLength
     def to_csv
-      CSV.generate do |csv|
-        csv << ['Report Timeframe', "#{time_range.begin} to #{time_range.end}"]
-        csv << ['Report Generated', Date.today.to_s] # rubocop:disable Rails/Date
-        csv << ['Issuer', issuers.join(', ')]
-        csv << []
-        csv << ['Metric', 'Number of accounts', '% of total from start']
-        csv << [
-          'New Users Started IAL1 Verification',
-          email_confirmation,
-          format_as_percent(numerator: email_confirmation, denominator: email_confirmation),
-        ]
-
-        csv << [
-          'New Users Completed IAL1 Password Setup',
-          two_fa_setup_visited,
-          format_as_percent(numerator: two_fa_setup_visited, denominator: email_confirmation),
-        ]
-
-        csv << [
-          'New Users Completed IAL1 MFA',
-          user_fully_registered,
-          format_as_percent(numerator: user_fully_registered, denominator: email_confirmation),
-        ]
-        csv << [
-          'New IAL1 Users Consented to Partner',
-          sp_redirect_initiated_new_users,
-          format_as_percent(
-            numerator: sp_redirect_initiated_new_users,
-            denominator: email_confirmation,
-          ),
-        ]
-        csv << []
-        csv << ['Total # of IAL1 Users', sp_redirect_initiated_all]
-        csv << []
-        csv << [
-          'AAL2 Authentication Requests from Partner',
-          oidc_auth_request,
-          format_as_percent(numerator: oidc_auth_request, denominator: oidc_auth_request),
-        ]
-        csv << [
-          'AAL2 Authenticated Requests',
-          sp_redirect_initiated_after_oidc,
-          format_as_percent(
-            numerator: sp_redirect_initiated_after_oidc,
-            denominator: oidc_auth_request,
-          ),
-        ]
+      as_tables.map do |table|
+        CSV.generate do |csv|
+          table.each do |row|
+            csv << row
+          end
+        end
```

Comment:
I would have `to_csv` return a single string, or call this `to_csvs` or `to_csv_array` or something